ETL/s3_to_dw_load.py

```diff
@@ -0,0 +1,111 @@
+import boto3
+import pandas as pd
+from sqlalchemy import create_engine, text
+import datetime
+import os
+import psycopg2
+import numpy as np
+
+db_url = os.environ["YDB_URL"]
+engine = create_engine(db_url)
+
+
+def check_no_rows_dw(conn):
+    query = """
+    SELECT *
+    FROM user_dim
+    LIMIT 1
+    """
+    result = conn.execute(text(query)).fetchall()
+    return result
+
+
+def get_latest_snapshot_datalake(days=1):
+    snap_dir = get_path_snapshot(days)
+    s3_client = boto3.client("s3")
+    s3_bucket = "analytics-ninjas"
+    s3_object_key = snap_dir.lstrip("/")
+    s3_response = s3_client.get_object(Bucket=s3_bucket, Key=s3_object_key)
+    df = pd.read_csv(s3_response["Body"])
+    return df
+
+
+def get_path_snapshot(days):
+    today = str(datetime.datetime.now() - datetime.timedelta(days=days))[0:10].replace(
+        "-", ""
+    )
+    return f"/stock_db/user/partition={today}/user.csv"
+
+
+def element_to_string(element):
+    return f"{element}"
+
+
+def get_records(df, records_type):
+    if records_type != "All":
+        df = df.loc[df["status"] == records_type]
+    df.rename(columns={"udated_at": "start_date"})
+    df = df.drop(columns=["status"])
+    df["end_date"] = "9999-01-01 00:00:00"
+    df["is_current"] = True
+    vectorized_element_to_string = np.vectorize(element_to_string)
+    records = ",".join(
+        vectorized_element_to_string(pd.Series(df.to_records(index=False).tolist()))
+    )
+    return df, records
+
+
+def update_end_date_outdated_records(conn, df):
+    update_values = get_records(df, "U")[1]
+    query = f"""
+    INSERT INTO user_dim (
+        user_id, name, email, start_date, end_date, is_current
+    ) 
+    VALUES {update_values}
+    ON CONFLICT (user_id, end_date) DO 
+    UPDATE 
+    SET 
+        end_date = EXCLUDED.start_date - interval '1 second',
+        is_current = false 
```

Comment:
What is the smallest unit of time for this? is second small enough?
Can it be `interval '1 millisecond'`?